packages/terser/src/index.js

```diff
@@ -22,48 +22,65 @@ function log_error(...m) {
-// input/output arguments. See discussion: https://github.com/bazelbuild/rules_nodejs/issues/822
-
-const inputs = argv.slice(0, outputArgIndex);
-const output = argv[outputArgIndex + 1];
-const residual = argv.slice(outputArgIndex + 2);
-
-// user override for the terser binary location. used in testing.
-const TERSER_BINARY = process.env.TERSER_BINARY || require.resolve('terser/bin/uglifyjs')
-// choose a default concurrency of the number of cores -1 but at least 1.
-const TERSER_CONCURENCY = (process.env.TERSER_CONCURRENCY || os.cpus().length - 1) || 1
-
-log_verbose(`Running terser/index.js
-  inputs: ${inputs}
-  output: ${output}
-  residual: ${residual}`);
+/**
+ * @param {string[]} residualArgs
+ * @param {string} output
+ */
+function processSourceMapArg(residualArgs, output) {
+  // copy args so we don't accidently mutate and process the same set of args twice
+  const argsCopy = [...residualArgs];
+  const sourceMapIndex = argsCopy.findIndex(a => a === '--source-map');
+  if (sourceMapIndex != -1) {
+    // the options for the source map arg is the next one
+    // if it changes in terser_minified.bzl this needs to be updated
+    const sourceMapOptsIndex = sourceMapIndex + 1
+    const sourceMapOpts = argsCopy[sourceMapOptsIndex];
+
+    // decode the options to find the base url option
+    // which we need below to compute the relative url
+    const baseKeyValue =
+        sourceMapOpts.split(',').map(g => g.split('=')).find(([key, value]) => key === 'base');
+    if (baseKeyValue) {
+      const [_, baseUrl] = baseKeyValue;
+
+      // we need to remove the baseUrl from output
+      // since output is from the execroot, but we need a workspace relative url
+      // so we replace the baseUrl, then remove the beginnning /
+      // since a absolute url is invalid for terser
+      const relativeUrl = output.replace(baseUrl, '').slice(1);
+      const modifiedOptions = sourceMapOpts + `,url=${relativeUrl}.map`;
+
+      argsCopy[sourceMapOptsIndex] = modifiedOptions;
+    } else {
+      throw new Error(
+          `[terser/index.js]: source map option "base" is required with the --source-map argument, options that were provided: ${
+              sourceMapOpts}`)
+    }
+  }
+
+
+  return argsCopy;
+}
 
 function isDirectory(input) {
   return fs.lstatSync(path.join(process.cwd(), input)).isDirectory();
 }
 
-function terserDirectory(input) {
+function terserDirectory(input, output, residual, terserBinary) {
   if (!fs.existsSync(output)) {
     fs.mkdirSync(output);
   }
 
+  const TERSER_CONCURENCY = (process.env.TERSER_CONCURRENCY || os.cpus().length - 1) || 1
+
   let work = [];
   let active = 0;
   let errors = [];
 
   function exec([inputFile, outputFile]) {
     active++;
-    let args = [TERSER_BINARY, inputFile, '--output', outputFile, ...residual];
+    let args =
+        [terserBinary, inputFile, '--output', outputFile, ...processSourceMapArg(residual, output)];
```

Comment:
Should be `...processSourceMapArg(residual, outputFile)` right?

Reply:
yes! must have messed that up in a rebase or something